source/time_tables/departure_boards.cpp

```diff
@@ -270,11 +325,82 @@ void departure_board(PbCreator& pb_creator, const std::string& request,
         map_route_stop_point[route_point] = stop_times;
     }
 
-    render(pb_creator, response_status, map_route_stop_point, handler.date_time, handler.max_datetime,
-              calendar_id, depth);
+    render(pb_creator,
+           response_status,
+           map_route_stop_point,
+           map_route_first_and_last_point,
+           handler.date_time,
+           handler.max_datetime,
+           calendar_id, depth);
 
     pb_creator.make_paginate(total_result, start_page, count, std::max(pb_creator.departure_boards_size(),
                                                                        pb_creator.stop_schedules_size()));
 }
+
+std::pair<DateTime, DateTime> shift_next_stop_time_to_opening_time(const routing::datetime_stop_time& next_stop_time,
+                                                                   const uint32_t opening_time)
+{
+    uint nb_days = DateTimeUtils::date(next_stop_time.first); // Nb days from the data set origin until the next stop time.
+
+    // sometimes a departure_time(stop_time) may exceed 24H and we suppose it doesn't exceed 48H  
+    uint32_t dep_time = math_mod(next_stop_time.second->departure_time, DateTimeUtils::SECONDS_PER_DAY);
+    if (dep_time < opening_time) {
+        return std::make_pair((nb_days - 1)*DateTimeUtils::SECONDS_PER_DAY + opening_time - 1,
```

Comment:
i have a test for `nb_days == 0`: https://github.com/CanalTP/navitia/pull/2509/files#diff-b4158732da04c6a27c4538fc599e1ce1R1577 and `nb_day == 365` since everything datetime is relative to the beginning of the production, I think it's ok